### PR TITLE
[Color Area] Dispatch change event when using keyboard

### PR DESCRIPTION
**Pull request by najikahalsema**
## Description

Similar to the other colour elements, ColorArea was not correctly dispatching change events when using the keyboard. That is now fixed. Hooray. Also, altering other tests that were previously incorrect/not optimal.

## Related Issue

fixes https://github.com/adobe/spectrum-web-components/issues/1469 by extension

## Motivation and Context

Matching functionality with other colour elements and providing accessibility for users 

## How Has This Been Tested?

Manually and with tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/color-area/src/ColorArea.ts [hunk 1]
```diff
@@ -247,10 +247,34 @@ export class ColorArea extends SpectrumElement {
...
+        if (deltaX != 0) {
             this.inputX.focus();
-        } else if (deltaY) {
+            this.inputX.dispatchEvent(
+                new Event('input', {
+                    bubbles: true,
+                    composed: true,
+                })
+            );
+            // this.inputX.dispatchEvent(
```
Comment (Westbrook) [inline packages/color-area/src/ColorArea.ts:258]: I think if we `event.preventDefault();` in line 213, to make this work a little more like `ColorSlider` then we can uncomment these `change` events.

If not, we should remove them.

Also, we should join the dispatches under a single conditional after the focus logic, something like:
```
if (deltaX != 0 || deltaY != 0) {
   // dispatch here, so it's in one place
}
```

File: packages/color-area/test/color-area.test.ts [hunk 2]
```diff
@@ -160,13 +161,16 @@ describe('ColorArea', () => {
...
+        expect(el.x).to.equal(0.67);
         expect(el.y).to.equal(0.25);
     });
     it('accepts "Arrow*" keypresses with alteration', async () => {
         const el = await fixture<ColorArea>(
             html`
-                <sp-color-area color="hsla(100, 50%, 50%, 1)"></sp-color-area>
+                <sp-color-area
+                    color="hsla(100, 50%, 50%, 1)"
+                    style="--spectrum-global-dimension-size-2400:192px"
```
Comment (Westbrook) [inline packages/color-area/test/color-area.test.ts:172]: Can you check if we ended up needing this?

File: packages/color-area/test/color-area.test.ts [hunk 3]
```diff
@@ -325,6 +333,82 @@ describe('ColorArea', () => {
...
+        el.inputY.dispatchEvent(
+            new Event('input', {
+                bubbles: true,
+                composed: true,
+            })
+        );
+        el.inputY.dispatchEvent(
+            new Event('change', {
+                bubbles: true,
+                composed: false,
```
Comment (Westbrook) [inline packages/color-area/test/color-area.test.ts:380]: Can you comment here too: `// native change events do not compose themselves by default`, it's just such a finicky reality...

File: packages/color-area/test/color-area.test.ts [hunk 4]
```diff
@@ -325,6 +333,82 @@ describe('ColorArea', () => {
...
+                ></sp-color-area>
+            `
+        );
+
+        await elementUpdated(el);
+
+        el.inputX.focus();
+
+        await sendKeys({ press: 'ArrowRight' });
+        await sendKeys({ press: 'ArrowRight' });
```
Comment (Westbrook) [inline packages/color-area/test/color-area.test.ts:405]: Might be worth doing all four directions here, for "completeness"?

**Review by Westbrook — CHANGES_REQUESTED** [review]
This looks great!

A couple of double checks to make sure we're not leaving extra code laying around and we'll be good.

Also, the Github Actions failure is their fault, not ours.

File: packages/color-area/test/color-area.test.ts (hunk 2 above)
Comment (najikahalsema) [reply packages/color-area/test/color-area.test.ts:172]: Doesn't seem like it.

File: packages/color-area/src/ColorArea.ts (hunk 1 above)
Comment (najikahalsema) [reply packages/color-area/src/ColorArea.ts:258]: I did check this out, and it didn't seem to work the way it was expected. I removed the change events for this reason.

File: packages/color-area/test/color-area.test.ts [hunk 5]
```diff
@@ -325,6 +330,112 @@ describe('ColorArea', () => {
...
+
+        await elementUpdated(el);
+
+        expect(inputSpy.callCount).to.equal(4);
+        expect(changeSpy.callCount).to.equal(2);
```
Comment (Westbrook) [inline packages/color-area/test/color-area.test.ts:407]: These should match, I think. One input and one change event for each key listened to... can you check what's causing this? Seems like maybe the `event.preventDefault()` note I sent before, but possibly something else.

Comment (Westbrook) [reply packages/color-area/test/color-area.test.ts:407]: For reference: https://github.com/adobe/spectrum-web-components/blob/main/packages/color-slider/test/color-slider.test.ts#L80-L95

**Review by Westbrook — APPROVED** [review]
Thanks so much for this. Looks like the branch is a little behind `main` but once that's caught up, you're good to go!

**Review by Westbrook — APPROVED** [review]

**Pull request merged**